libdnf-cli/utils/tty.hpp

```diff
@@ -0,0 +1,37 @@
+#ifndef LIBDNF_CLI_UTILS_TTY_TTY_HPP
+#define LIBDNF_CLI_UTILS_TTY_TTY_HPP
+
+
+#include <iostream>
+
+
+namespace libdnf::cli::utils::tty {
+
+
+std::ostream & reset(std::ostream & stream);
+
+std::ostream & bold(std::ostream & stream);
+std::ostream & underline(std::ostream & stream);
+std::ostream & blink(std::ostream & stream);
+
+std::ostream & black(std::ostream & stream);
+std::ostream & red(std::ostream & stream);
+std::ostream & green(std::ostream & stream);
+std::ostream & yellow(std::ostream & stream);
+std::ostream & blue(std::ostream & stream);
+std::ostream & magenta(std::ostream & stream);
+std::ostream & cyan(std::ostream & stream);
+std::ostream & white(std::ostream & stream);
+
+std::ostream & clear_line(std::ostream & stream);
+std::ostream & cursor_up(std::ostream & stream);
+
+
+int get_width();
+bool is_interactive(void);
```

Comment:
`void` argument can be used in C++ however it is superfluous (C++ standard: 8.3.5.4). Empty parameter-declaration-clause is usually used for functions without arguments in C++. 2. Declaration of `bool is_interactive(void);` is inconsistent with previous declaration `int get_width();`.

Note:
Beware of the `C` language. It's different there. Always use `void` for functions without a parameter in `C` (C standard: 6.7.6.3.10). Using empty list has other meaning (C standard: 6.7.6.3.14).

### C++: International Standard, N4296 Working Draft, Standard for Programming Language C ++

8.3.5.4
The parameter-declaration-clause determines the arguments that can be specified, and their processing, when
the function is called. [ Note: the parameter-declaration-clause is used to convert the arguments specified
on the function call; see 5.2.2. — end note ] **If the parameter-declaration-clause is empty, the function takes
no arguments. A parameter list consisting of a single unnamed parameter of non-dependent type void is
equivalent to an empty parameter list.** Except for this special case, a parameter shall not have type cv void...

### C: INTERNATIONAL STANDARD ©ISO/IEC ISO/IEC 9899:201x

6.7.6.3.10
The special case of an unnamed parameter of type `void` as the only item in the list
specifies that the function has no parameters.

6.7.6.3.14
An identifier list declares only the identifiers of the parameters of the function. An empty
list in a function declarator that is part of a definition of that function specifies that the
function has no parameters. The empty list in a function declarator that is not part of a
definition of that function specifies that no information about the number or types of the
parameters is supplied.

Reply:
Thanks for the verbose explanation, quite interesting details.
I've fixed the code by removing `void`.